### ISSUE TITLE
[BUG]: Bing (search engine) is broken

### ISSUE THREAD

**Issue by MohamedMehrath** [undergroundwires/privacy.sexy#329]
### Description

I used privacy.sexy (amazing tool btw) and bing isn't working well,  When I click on Copilot, nothing happens. Bottom righthand side of screen shows javascript:void(0) also Bing isn't showing pictures.

### OS

Windows 10 Pro

### Screenshots

![image](https://github.com/undergroundwires/privacy.sexy/assets/91329167/90d95eb6-1aa4-42e3-ade7-65c5687414ec)
![image](https://github.com/undergroundwires/privacy.sexy/assets/91329167/4edc43da-0de3-4180-97ee-783e62db1b8d)

**Comment by undergroundwires** [comment]
This is most likely related to a host entry under `Block tracking hosts` category. Try to revert those and please share if you figure out which host exactly is blocking copilot as side-effect so we can document/discourage or remove that host.

**Comment by MohamedMehrath** [comment]
I reverted all `Block tracking hosts` and re-enabled each one, and I ended up enabling all of them again, fixing the problem.

**Comment by undergroundwires** [comment]
Thank you for the insights.

I cannot reproduce this. But it would be appreciated if you could test which host is causing this exactly (which bing host), so we can exclude it from privacy.sexy or document the behavior.

I cannot test this because for me, Copilot on Edge through Bing works when all hosts including  "Block Cortana and Live Tiles hosts" are blocked:

![image](https://github.com/undergroundwires/privacy.sexy/assets/15555035/5a1d87da-d4a9-4025-80ad-84d12b7d92b6)

**Comment by maxysoft** [comment]
There is `r.bing.com` that needs to be removed from hosts to make copilot works

**Comment by undergroundwires** [comment]
Thank you for figuring this out and reporting back @maxysoft.

`r.bing.com` is now excluded with the added documentation (see at the end):

![image](https://github.com/undergroundwires/privacy.sexy/assets/15555035/6f956944-b2b5-4964-892e-55848cc2627f)

It will be released in next patch.

**Comment by joethesaint** [comment]
Now that it has been excluded, how then can we that are facing the issue fix it?

**Comment by undergroundwires** [comment]
Here's the revert code for `r.bing.com`:

```cmd
PowerShell -ExecutionPolicy Unrestricted -Command "$domain ='r.bing.com'; $hostsFilePath = "^""$env:WINDIR\System32\drivers\etc\hosts"^""; $comment = "^""managed by privacy.sexy"^""; $hostsFileEncoding = [System.Text.Encoding]::UTF8; $blockingHostsEntries = @(; @{ AddressType = "^""IPv4"^"";  IPAddress = '0.0.0.0'; }; @{ AddressType = "^""IPv6"^"";  IPAddress = '::1'; }; ); try {; $isHostsFilePresent = Test-Path -Path $hostsFilePath -PathType Leaf -ErrorAction Stop; } catch {; Write-Error "^""Failed to check hosts file existence. Error: $_"^""; exit 1; }; if (-Not $isHostsFilePresent) {; Write-Output 'Skipping, the hosts file does not exist.'; exit 0; }; foreach ($blockingEntry in $blockingHostsEntries) {; Write-Output "^""Processing removal for $($blockingEntry.AddressType) entry."^""; try {; $hostsFileContents = [System.IO.File]::ReadAllText($hostsFilePath, $hostsFileEncoding); } catch {; Write-Error "^""Failed to read the hosts file for removal. Error: $_"^""; continue; }; $hostsEntryLine = "^""$($blockingEntry.IPAddress)`t$domain $([char]35) $comment"^""; if ([String]::IsNullOrWhiteSpace($hostsFileContents) -Or (-Not $hostsFileContents.Contains($hostsEntryLine))) {; Write-Output 'Skipping, entry not found.'; continue; }; $hostsEntryRemovalPattern = [regex]::Escape($hostsEntryLine) + "^""(\r?\n)?"^""; $hostsFileContentAfterRemoval = $hostsFileContents -Replace $hostsEntryRemovalPattern, "^"""^""; try {; [System.IO.File]::WriteAllText($hostsFilePath, $hostsFileContentAfterRemoval, $hostsFileEncoding); Write-Output 'Successfully removed the entry.'; } catch {; Write-Error "^""Failed to remove the entry. Error: $_"^""; continue; }; }"
```

**Comment by undergroundwires** [comment]
Here's the revert code for `r.bing.com`:

```batchfile
PowerShell -ExecutionPolicy Unrestricted -Command "$domain ='r.bing.com'; $hostsFilePath = "^""$env:WINDIR\System32\drivers\etc\hosts"^""; $comment = "^""managed by privacy.sexy"^""; $hostsFileEncoding = [System.Text.Encoding]::UTF8; $blockingHostsEntries = @(; @{ AddressType = "^""IPv4"^"";  IPAddress = '0.0.0.0'; }; @{ AddressType = "^""IPv6"^"";  IPAddress = '::1'; }; ); try {; $isHostsFilePresent = Test-Path -Path $hostsFilePath -PathType Leaf -ErrorAction Stop; } catch {; Write-Error "^""Failed to check hosts file existence. Error: $_"^""; exit 1; }; if (-Not $isHostsFilePresent) {; Write-Output 'Skipping, the hosts file does not exist.'; exit 0; }; foreach ($blockingEntry in $blockingHostsEntries) {; Write-Output "^""Processing removal for $($blockingEntry.AddressType) entry."^""; try {; $hostsFileContents = [System.IO.File]::ReadAllText($hostsFilePath, $hostsFileEncoding); } catch {; Write-Error "^""Failed to read the hosts file for removal. Error: $_"^""; continue; }; $hostsEntryLine = "^""$($blockingEntry.IPAddress)`t$domain $([char]35) $comment"^""; if ([String]::IsNullOrWhiteSpace($hostsFileContents) -Or (-Not $hostsFileContents.Contains($hostsEntryLine))) {; Write-Output 'Skipping, entry not found.'; continue; }; $hostsEntryRemovalPattern = [regex]::Escape($hostsEntryLine) + "^""(\r?\n)?"^""; $hostsFileContentAfterRemoval = $hostsFileContents -Replace $hostsEntryRemovalPattern, "^"""^""; try {; [System.IO.File]::WriteAllText($hostsFilePath, $hostsFileContentAfterRemoval, $hostsFileEncoding); Write-Output 'Successfully removed the entry.'; } catch {; Write-Error "^""Failed to remove the entry. Error: $_"^""; continue; }; }"
```

**Comment by undergroundwires** [comment]
The fix is released since [0.13.3](https://github.com/undergroundwires/privacy.sexy/releases/edit/0.13.3) 🎉.

Here's the revert code for `r.bing.com` for those who wants to allow `r.bing.com` back (@joethesaint):

```batchfile
PowerShell -ExecutionPolicy Unrestricted -Command "$domain ='r.bing.com'; $hostsFilePath = "^""$env:WINDIR\System32\drivers\etc\hosts"^""; $comment = "^""managed by privacy.sexy"^""; $hostsFileEncoding = [System.Text.Encoding]::UTF8; $blockingHostsEntries = @(; @{ AddressType = "^""IPv4"^"";  IPAddress = '0.0.0.0'; }; @{ AddressType = "^""IPv6"^"";  IPAddress = '::1'; }; ); try {; $isHostsFilePresent = Test-Path -Path $hostsFilePath -PathType Leaf -ErrorAction Stop; } catch {; Write-Error "^""Failed to check hosts file existence. Error: $_"^""; exit 1; }; if (-Not $isHostsFilePresent) {; Write-Output 'Skipping, the hosts file does not exist.'; exit 0; }; foreach ($blockingEntry in $blockingHostsEntries) {; Write-Output "^""Processing removal for $($blockingEntry.AddressType) entry."^""; try {; $hostsFileContents = [System.IO.File]::ReadAllText($hostsFilePath, $hostsFileEncoding); } catch {; Write-Error "^""Failed to read the hosts file for removal. Error: $_"^""; continue; }; $hostsEntryLine = "^""$($blockingEntry.IPAddress)`t$domain $([char]35) $comment"^""; if ([String]::IsNullOrWhiteSpace($hostsFileContents) -Or (-Not $hostsFileContents.Contains($hostsEntryLine))) {; Write-Output 'Skipping, entry not found.'; continue; }; $hostsEntryRemovalPattern = [regex]::Escape($hostsEntryLine) + "^""(\r?\n)?"^""; $hostsFileContentAfterRemoval = $hostsFileContents -Replace $hostsEntryRemovalPattern, "^"""^""; try {; [System.IO.File]::WriteAllText($hostsFilePath, $hostsFileContentAfterRemoval, $hostsFileEncoding); Write-Output 'Successfully removed the entry.'; } catch {; Write-Error "^""Failed to remove the entry. Error: $_"^""; continue; }; }"
```

**Comment by joethesaint** [comment]
Thank you, I noticed bing is working after updating, did not get to revert
at the end of the day
